src/plugin/index.ts

```diff
@@ -1,15 +1,10 @@
 import { hasHOC } from './utils'
 
-const dir =
-  process.env.NX_WORKSPACE_ROOT ||
-  process.env.LERNA_PACKAGE_NAME ||
-  process.env.NEXT_APP_PATH ||
-  process.cwd()
-
 export default function nextTranslate(nextConfig: any = {}) {
   const fs = require('fs')
   const path = require('path')
   const test = /\.(tsx|ts|js|mjs|jsx)$/
+  const dir = process.env.NEXT_APP_PATH || nextConfig.dir || process.cwd()
```

Comment:
@aralroca , this version solved https://github.com/vinissimus/next-translate/issues/395#issuecomment-744550219.

Just notice, **dir** is not a *nextConfig* property. See https://github.com/vercel/next.js/blob/977b3ed7698e79062db9a4f76dddb5c377b286bb/packages/next/next-server/server/config.ts#L32. Maybe it is not a good idea to extend that object, it can led to conflicts in future.

Reply:
@dapuicon So what do you propose? Maybe rename "dir" to "nextTranslateDir" and clean it up inside our plugin?

Reply:
@aralroca , that is an option. I think that It depends on the taste on each one. I don't like black magic so I prefer something like this

```ts
export default function nextTranslate({ nextConfig = {}, nextTranslateConfig = {}}: any) {
  const fs = require('fs')
  const path = require('path')
  const test = /\.(tsx|ts|js|mjs|jsx)$/
  const dir = process.env.NEXT_APP_PATH || nextTranslateConfig.dir || process.cwd()
  const arePagesInsideSrc = fs.existsSync(path.join(dir, 'src/pages'))
```

Reply:
I'm thinking about it, and I don't like that with this change now there are two different configurations (i18n.js and here). Maybe keeping only the environment variable to define this might make sense? Maybe with a better name, like `NEXT_TRANSLATE_PATH`, together with `path.join(process.cwd(), process.env.NEXT_TRANSLATE_PATH)` to define It in as relative

Reply:
I agree two different configurations is not a good idea. However I don't like having an environment variable with a relative path.
I would prefer keeping the environment variable but with an abosulte path.

```js
const dir = process.env.NEXT_TRANSLATE_PATH || process.cwd()
```

¿what do you think?

Reply:
With:

```js
 const dir = path.resolve(path.relative(process.cwd(), process.env.NEXT_TRANSLATE_PATH || '.'))
```

it supports both, relative and absolute 😊

Reply:
@dapuicon I did a new PR: https://github.com/vinissimus/next-translate/pull/409